### DISCUSSION TITLE
can't redefine non-configurable property "$nextTick"

### DISCUSSION THREAD

**Discussion by 0x44454c** [alpinejs/alpine#2107] [1. Help] [answered]
```
<script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3/dist/cdn.min.js"></script>

<div x-data="info">
  <span x-text="data"></span>
</div>

<div x-data="info">
  <span x-text="data"></span>
</div>

<script>
  let info ={
  data: "hello"
}

document.addEventListener("alpine:init", () => {
    info = Alpine.reactive(info)
    Alpine.data("info", () => info);
 });
</script>
```

Above code gives following error:

![image](https://user-images.githubusercontent.com/58584472/134410248-a25c437f-8627-454f-9c2b-f1d30b6c39a5.png)

How can I solve the issue?

**Comment by SimoTod** [comment, marked as answer]
You don't need `info = Alpine.reactive(info)`.
Alpine data is already reactive, you can see an example in the documentation

**Reply by SimoTod** [reply]
Also (which is probably the cause of the error) Alpine.data should return a fresh object every time, you are always returning the same window.info. If you want to share state between components, you should use Alpine.store.

**Reply by 0x44454c** [reply]
`info = Alpine.reactive(info)` has been used to update the value from another javascript function which has no connection with alpine. :shrug: Basically to get more control over the object.

**Reply by SimoTod** [reply]
That pattern would not work, as mentioned above Alpine.data should not return the same object every time. X-data will inject magic methods into the object so if you do it twice you get that error. Note that, if it didn't throw an error, the megic methods wouldn't work anyway because they get passed one of the context which would be wrong for the other one.
You definitely want to use a store: https://codepen.io/SimoTod/pen/wvexRLW

**Reply by 0x44454c** [reply]
Well, it worked but sometimes it throws error that Alpine isn't defined due to defer attribute in script tag. Also, it's my personal opinion that `Alpine.store` is too much verbose for getting a setting a value for many times.

**Reply by SimoTod** [reply]
Yuo need to wrap your code in an event listener and wait for the alpine:init event like the example in the docs.

If you find the assignment verbose, just use a variable `const mystore = Alpine.store('foo')`

**Reply by 0x44454c** [reply]
I wrapped my whole code around a function a called it on `alpine:init`, also refactored my code to use Alpine store. All thanks to you. 

Oh by the way I recently got to know that if I make an Object reactive once before declaring it in `Alpine.data`, then my setting and getting on the object also affects DOM. 🤷🏻‍♂️ 

Anyway Thank you again.
